icu4c/source/common/ucnvmbcs.cpp

```diff
@@ -3144,13 +3144,14 @@ ucnv_MBCSGetNextUChar(UConverterToUnicodeArgs *pArgs,
     }
 
     if(c<0) {
-        if(U_SUCCESS(*pErrorCode) && source==sourceLimit && lastSource<source) {
+        if(U_SUCCESS(*pErrorCode) && source==sourceLimit && lastSource<source &&
+           size_t(source-lastSource) <= sizeof(cnv->toUBytes)) {
```

Comment:
```suggestion
        if(U_SUCCESS(*pErrorCode) && source==sourceLimit && lastSource<source &&
                (size_t)(source-lastSource) <= sizeof(cnv->toUBytes)) {
```

Reply:
suggesting style fixes -- but i am fairly sure that this condition is supposed to always be true; don't have time to dig into details, so ok with adding it